convert-hf-to-gguf.py

```diff
@@ -281,6 +283,8 @@ def _get_model_architecture(self) -> gguf.MODEL_ARCH:
             return gguf.MODEL_ARCH.NOMIC_BERT
         if arch == "GemmaForCausalLM":
             return gguf.MODEL_ARCH.GEMMA
+        if arch == "Starcoder2ForCausalLM":
+            return gguf.MODEL_ARCH.STARCODER2
```

Comment:
This appears to be the first and only architecture that does not subclass Model. Could you add a comment to `from_model_architecture` to make this clearer?

Reply:
Added redundant conditional similar to `Qwen2ForCausalLM` model case. This should resolve the above.

Reply:
That works. Btw, I think this function should probably be changed to a dict lookup at some point, especially if we want to explicitly list each arch string like this.